### ISSUE TITLE
protobuf-kotlin 3.20 duplicate classes with protobuf-java

### ISSUE THREAD

**Issue by maikelvdh** [protocolbuffers/protobuf#9902]
**What version of protobuf and what language are you using?**
Version: 3.20.0
Language: Kotlin

**What operating system (Linux, Windows, ...) and version?**
macOS 12.2.1

**What runtime / compiler are you using (e.g., python version or gcc version)**
Protoc 3.20.0
Gradle 7.4.2

**What did you do?**

    1. Create a new Android studio project
    2. Add dependency on com.google.protobuf:protobuf-kotlin:3.20.0
    3. Build project

**What did you expect to see**
Project builds successfully

**What did you see instead?**
Gradle outputs the following set of errors (reduced the size):

```
Duplicate class com.google.protobuf.Any found in modules protobuf-java-3.20.0 (com.google.protobuf:protobuf-java:3.20.0) and protobuf-kotlin-3.20.0 (com.google.protobuf:protobuf-kotlin:3.20.0)
Duplicate class com.google.protobuf.Any$1 found in modules protobuf-java-3.20.0 (com.google.protobuf:protobuf-java:3.20.0) and protobuf-kotlin-3.20.0 (com.google.protobuf:protobuf-kotlin:3.20.0)
Duplicate class com.google.protobuf.Any$Builder found in modules protobuf-java-3.20.0 (com.google.protobuf:protobuf-java:3.20.0) and protobuf-kotlin-3.20.0 (com.google.protobuf:protobuf-kotlin:3.20.0)
Duplicate class com.google.protobuf.AnyOrBuilder found in modules protobuf-java-3.20.0 (com.google.protobuf:protobuf-java:3.20.0) and protobuf-kotlin-3.20.0 (com.google.protobuf:protobuf-kotlin:3.20.0)
Duplicate class com.google.protobuf.AnyProto found in modules protobuf-java-3.20.0 (com.google.protobuf:protobuf-java:3.20.0) and protobuf-kotlin-3.20.0 (com.google.protobuf:protobuf-kotlin:3.20.0)
Duplicate class com.google.protobuf.Api found in modules protobuf-java-3.20.0 (com.google.protobuf:protobuf-java:3.20.0) and protobuf-kotlin-3.20.0 (com.google.protobuf:protobuf-kotlin:3.20.0)
Duplicate class com.google.protobuf.Api$1 found in modules protobuf-java-3.20.0 (com.google.protobuf:protobuf-java:3.20.0) and protobuf-kotlin-3.20.0 (com.google.protobuf:protobuf-kotlin:3.20.0)
Duplicate class com.google.protobuf.Api$Builder found in modules protobuf-java-3.20.0 (com.google.protobuf:protobuf-java:3.20.0) and protobuf-kotlin-3.20.0 (com.google.protobuf:protobuf-kotlin:3.20.0)
Duplicate class com.google.protobuf.ApiOrBuilder found in modules protobuf-java-3.20.0 (com.google.protobuf:protobuf-java:3.20.0) and protobuf-kotlin-3.20.0 (com.google.protobuf:protobuf-kotlin:3.20.0)
Duplicate class com.google.protobuf.ApiProto found in modules protobuf-java-3.20.0 (com.google.protobuf:protobuf-java:3.20.0) and protobuf-kotlin-3.20.0 (com.google.protobuf:protobuf-kotlin:3.20.0)
Duplicate class com.google.protobuf.BoolValue found in modules protobuf-java-3.20.0 (com.google.protobuf:protobuf-java:3.20.0) and protobuf-kotlin-3.20.0 (com.google.protobuf:protobuf-kotlin:3.20.0)
Duplicate class com.google.protobuf.BoolValue$1 found in modules protobuf-java-3.20.0 (com.google.protobuf:protobuf-java:3.20.0) and protobuf-kotlin-3.20.0 (com.google.protobuf:protobuf-kotlin:3.20.0)
```

As the error suggests, this is caused because both protobuf-kotlin and protobuf-java include the same generated sources from proto files. This seems to be introduced with following change: https://github.com/protocolbuffers/protobuf/commit/a32a2a3b57e82b4e97d00a93b6e37773b7fdd763

**Comment by deannagarcia** [comment]
Thanks for this report! Working on a PR to fix it now.

**Comment by deannagarcia** [comment]
Should be fixed by #9909, let me know if you still see this problem.
